src/Html.php

```diff
@@ -3474,6 +3482,29 @@ public static function computeGenericDateTimeSearch($val, $force_day = false, $s
             return date($format_use, mktime($hour, $minute, $second, $month, $day, $year));
         }
 
+       // Search on begin of month / year
+        if (strstr($val, 'END')) {
+            $hour   = 0;
+            $minute = 0;
+            $second = 0;
+            $month  = date("n", $specifictime);
+            $day    = 1;
+            $year   = date("Y", $specifictime);
+
+            switch ($val) {
+                case "ENDYEAR":
+                    $month = 12;
+                    $day = 31;
+                    break;
+
+                case "ENDMONTH":
+                    $day = date("t", $specifictime);
+                    break;
+            }
+
+            return date($format_use, mktime($hour, $minute, $second, $month, $day, $year));
+        }
```

Comment:
This should be mutualized with the `if (strstr($val, 'BEGIN')) {`.

```php
if (strstr($val, 'BEGIN') || strstr($val, 'END')) {
    ...
    switch ($val) {
        ...
        case "ENDYEAR":
            $month = 12;
            $day = 31;
            break;

        case "ENDMONTH":
            $day = date("t", $specifictime);
            break;
```

Reply:
Done in new code